cloud/resource_cloud_environment.go

```diff
@@ -293,8 +293,69 @@ func resourceCloudEnvironmentRead(ctx context.Context, d *schema.ResourceData, m
 }
 
 func resourceCloudEnvironmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	return diag.FromErr(fmt.Errorf("ERROR_UPDATE_CLOUD_ENVIRONMENT: " +
-		"The cloud environment does not support updates, please recreate it"))
+	namespace := d.Get("organization").(string)
+	waitForCompletion := d.Get("wait_for_completion").(bool)
+	name := strings.Split(d.Id(), "/")[1]
+
+	clientSet, err := getClientSet(getFactoryFromMeta(meta))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("ERROR_INIT_CLIENT_ON_UPDATE_CLOUD_ENVIRONMENT: %w", err))
+	}
+
+	if d.HasChanges("organization") ||
+		d.HasChanges("cloud_connection_name") ||
+		d.HasChanges("region") ||
+		d.HasChanges("network_id") ||
+		d.HasChanges("network_cidr") {
+		return diag.FromErr(fmt.Errorf("ERROR_UPDATE_CLOUD_ENVIRONMENT: " +
```

Comment:
Can we display the immutable fields in the error message, unless it may confuse the customer?